activate-polyfills guide (#1476)
TBD54566975/developer.tbd.website: site/docs/web5/build/decentralized-web-nodes/access-data-with-drls.mdx

---
sidebar_position: 10
---

import CodeSnippet from '@site/src/components/CodeSnippet';

> # Accessing Data with DRLs

Reviewer: ```suggestion
# Access Data with DRLs
```

A [Decentralized Web App (DWA)](/blog/dwas-vs-pwas) is a Progressive Web App (PWA) that incorporates Web5 functionalities, providing offline capabilities and offline synchronization of [Decentralized Web Nodes (DWNs)](/docs/web5/learn/decentralized-web-nodes/). 

This guide will walk you through how to read from DWNs within your DWA via **Decentralized Resource Locators (DRLs)**. For a practical implementation, refer to the [DWA starter app](https://github.com/TBD54566975/dwa-starter).

> ## Import packages

Reviewer: ```suggestion
## Import package
```

Import the following package to handle interactions with DWNs and manage offline data:

```js
import { activatePolyfills } from '@web5/api';
```

## Set up a service worker
[Service workers](https://web.dev/learn/pwa/service-workers) enable offline functionality and background processing.

The following snippet configures a service worker to activate polyfills, manage caching, and handle data synchronization tasks in the background:

```js
// Activate Web5 Service Worker Polyfills
activatePolyfills({
  onCacheCheck() {
    return {
      ttl: 30000,
    };
  },
});
```

## Fetch data via DRLs

> DRLs enable direct read access to permussioned data stored in DWNs. To read records from a DWN,  After writing records to a DWN, you can use DRLs (prefixed with `https://dweb`) to fetch the data. 

Reviewer: ```suggestion
Decentralized Resource Locators (**DRLs**) enable direct read access to permissioned data stored in DWNs. To read records from a DWN,  After writing records to a DWN, you can use DRLs (prefixed with `https://dweb`) to fetch the data. 
```
